src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -717,14 +688,15 @@ public ReadOnlyTensorSpan<T> Slice(params scoped ReadOnlySpan<NRange> ranges)
         public bool TryFlattenTo(scoped Span<T> destination)
         {
             bool retVal = false;
-            if (destination.Length < _flattenedLength)
+            if (destination.Length < _shape.FlattenedLength)
             {
                 scoped Span<nint> curIndexes;
                 nint[]? curIndexesArray;
-                if (Rank > 6)
+                if (Rank > TensorSpan.MaxRankForStackAlloc)
```

Comment:
Similar here, I don't think this is related to the inline storage limitations of `TensorShape` and could use the more general stackalloc upper limit of 1024 bytes per method.